pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -359,7 +359,7 @@ func TestReconcile_ExplicitDefaultSA(t *testing.T) {
 		tb.TaskRunServiceAccountName("test-sa"),
 	))
 	taskruns := []*v1beta1.TaskRun{taskRunSuccess, taskRunWithSaSuccess}
-	defaultSAName := "pipelines"
+	defaultSAName := "default"
```

Comment:
looking at the PR that added this change (https://github.com/tektoncd/pipeline/pull/1227) and trying to understand the intent, i think the goal was to explicitly set the serviceaccount to something other than default, is that your understanding as well? (or maybe it just needs to be something other than "tekton" https://github.com/tektoncd/pipeline/blob/master/pkg/apis/config/testdata/config-defaults.yaml#L22 ?)

it looks like the default value is "default" so i think by changing this to "default" we might be removing the thing this test is meant to cover

Reply:
yup its my understanding as well

This particular unit test is testing two things:

(1) taskrun without having specified service account in the spec unlike the [next test case](https://github.com/tektoncd/pipeline/blob/master/pkg/reconciler/taskrun/taskrun_test.go#L359) in the same function (2) changing default service account through config map

Trying to think if it's possible to separate these two.

And I am not sure if we could be removing the thing this test is meant to cover because default value is `tekton` according to `testdata` and irrespective of what it is set to (other than `tekton`), the same value is set in config-map and sent to controller. 🤔 

I am still looking into this and would appreciate input from config map experts 🙏

Reply:
I am working on getting this hard coded default out of `creds_init.go` 👩‍💻

Reply:
I have created PR https://github.com/tektoncd/pipeline/pull/3168 to remove this hard coded default and also add it to the pipelinerun/taskrun `spec`.

Reply:
I don't think the problem is the *hardcoded* `default` serviceaccount (although, it would be nice if it wasn't harcoded :stuck_out_tongue: )